### PR TITLE
Include widget configuration in the template

### PR DESCRIPTION
**Pull request by Hixie**
This should help new users figure out how to set up their constructors
and use config data from the build method of a State.

### CODE REVIEW

File: packages/flutter_tools/templates/create/lib/main.dart.tmpl [hunk 1]
```diff
@@ -30,18 +29,24 @@ class MyApp extends StatelessWidget {
...
       ),
-      home: new MyHomePage(),
+      home: new MyHomePage(title: 'Flutter Demo'),
     );
   }
 }
 
 class MyHomePage extends StatefulWidget {
-  MyHomePage({Key key}) : super(key: key);
+  MyHomePage({ Key key, this.title }) : super(key: key);
```
Comment (abarth) [inline packages/flutter_tools/templates/create/lib/main.dart.tmpl]: We've been using dartfmt style for the template so it doesn't change when the developer autoformats it.

**Comment by abarth** [comment]
LGTM

Looks like Travis found your errant style.

File: packages/flutter_tools/templates/create/lib/main.dart.tmpl (hunk 1 above)
Comment (Hixie) [inline packages/flutter_tools/templates/create/lib/main.dart.tmpl]: Aw man, I'm gonna have to get use to uglier named argument blocks aren't I.

File: packages/flutter_tools/templates/create/lib/main.dart.tmpl [hunk 2]
```diff
@@ -30,18 +29,24 @@ class MyApp extends StatelessWidget {
...
 
   // This widget is the home page of your application. It is stateful,
   // meaning that it has a State object (defined below) that contains
   // fields that affect how it looks.
 
+  // This class is the configuration for the state. It holds the
+  // values (in this case the title) provided by the parent (in this
+  // case the App widget) and used by the build method of the State.
```
Comment (Hixie) [inline packages/flutter_tools/templates/create/lib/main.dart.tmpl]: Yeah, good idea.

Comment (Hixie) [inline packages/flutter_tools/templates/create/lib/main.dart.tmpl]: Added ` // Fields in a Widget subclass are always marked "final".`

**Comment by Hixie** [comment]
will check in when travis goes green

**Pull request merged**